e2etests/pkg/config/config.go

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package config
+
+import (
+	"context"
+
+	frrk8sv1beta1 "github.com/metallb/frrk8s/api/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type Updater struct {
+	client client.Client
+}
+
+func NewUpdater(r *rest.Config) (*Updater, error) {
+	myScheme := runtime.NewScheme()
+	if err := frrk8sv1beta1.AddToScheme(myScheme); err != nil {
+		return nil, err
+	}
+
+	cl, err := client.New(r, client.Options{
+		Scheme: myScheme,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return &Updater{
+		client: cl,
+	}, nil
+}
+
+func (u *Updater) Update(configs ...frrk8sv1beta1.FRRConfiguration) error {
+	for i, config := range configs {
+		_, err := controllerutil.CreateOrUpdate(context.Background(), u.client, &config, func() error {
+			old := &configs[i].Spec
+			config.Spec = *old.DeepCopy()
+			return nil
```

Comment:
from the CreateOrUpdate doc:
```
// CreateOrUpdate creates or updates the given object in the Kubernetes                                                                                                                       
// cluster. The object's desired state must be reconciled with the existing                                                                                                                   
// state inside the passed in callback MutateFn. 
```

It means that when you are updating, it gives you the object you plan to update. In our case, we care only about changing the spec. If we don't do this, it will complain about generation and other fields being empty on update.